presto-druid/src/main/java/com/facebook/presto/druid/metadata/DruidSegmentInfo.java

```diff
@@ -112,11 +114,41 @@ public DeepStorageType getDeepStorageType()
         return DeepStorageType.valueOf(loadSpecification.get(DEEP_STORAGE_TYPE_KEY).toUpperCase());
     }
 
-    public String getDeepStoragePath()
+    public URI getDeepStoragePath() throws IOException
     {
-        Map<String, String> loadSpecification = getLoadSpecification()
+        // Logic from https://github.com/apache/druid/blob/63c1746fe40fc40d60dbc77f411e58949c102bfa/indexing-hadoop/src/main/java/org/apache/druid/indexer/JobHelper.java#L776
+        final Map<String, String> loadSpec = getLoadSpecification()
                 .orElseThrow(() -> new PrestoException(DRUID_METADATA_ERROR, format("Malformed segment loadSpecification: %s", getLoadSpecification())));
-        return loadSpecification.get(SEGMENT_PATH_KEY);
+        final String type = loadSpec.get(DEEP_STORAGE_TYPE_KEY);
+        final URI segmentLocURI;
+        switch (type) {
+            case "s3_zip":
+                final String s3schema = "s3a".equals(loadSpec.get("S3Schema")) ? "s3a" : "s3n";
```

Comment:
extract s3a, s3n, S2Schema and etc. as constants?

Reply:
done